### PR TITLE
fix(ui-react): Pagination current item a11y

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

Fixes: https://github.com/aws-amplify/amplify-ui/issues/1982

To improve the accessibility of the Pagination component:
- current item button is focusable
- screenreader announces if the item is current or not

https://user-images.githubusercontent.com/48109584/180571531-8cb72f58-70c6-4fbf-bb77-2ab775f5a9d0.mov

#### Issue #, if available

https://github.com/aws-amplify/amplify-ui/issues/1982

#### Description of how you validated changes

Ran the docs locally and used a screen reader. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated
- [X] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [X] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hbuchel — COMMENTED** [review]
Tiny tiny nit: What do you think about updating our default `currentPageLabel` (and docs under Accessibility section) to be:
```
currentPageLabel: 'Page:',
```
instead of `Current Page: `? It reduces the verbosity a little since the aria-current already reads out the word "current item". It would then announce:

**"Page 2, current item, button, group"**  instead of **"Current Page 2, current item, button group"**

**Review by hbuchel — APPROVED** [review]
LGTM! Thanks for the updates!

**Comment by hbuchel** [comment]
> LGTM! Thanks for the updates!

Oops spoke  too soon on this. Looks like there is a failing unit test.

**Review by hbuchel — APPROVED** [review]

File: packages/react/src/primitives/Pagination/PaginationItem.tsx [hunk 1]
```diff
@@ -60,7 +60,8 @@ export const PaginationItem: React.FC<PaginationItemProps> = ({
         <View as="li">
           {page === currentPage ? (
             <Flex
-              as="span"
+              aria-current="true"
```
Comment (reesscot) [inline packages/react/src/primitives/Pagination/PaginationItem.tsx:63]: @hbuchel Should this be set to `aria-current="page"`? 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current#values

**Review by reesscot — APPROVED** [review]
LGTM pending my question below

File: packages/react/src/primitives/Pagination/PaginationItem.tsx (hunk 1 above)
Comment (hbuchel) [reply packages/react/src/primitives/Pagination/PaginationItem.tsx:63]: Oops sorry, yeah that is better, thanks! @joebuono there is a comment we can remove just below it too that references the menu aria pattern that isn't relevant anymore.

**Review by hbuchel — APPROVED** [review]

**Review by 0618 — APPROVED** [review]
:shipit:

**Pull request merged**